server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocatorTests.java

```diff
@@ -95,93 +88,74 @@ public void testDecideShardAllocation() {
-                throw new UnsupportedOperationException("Not required for test");
-            }
-
-            @Override
-            @SuppressForbidden(reason = "This is required to test balancing by ingest load")
-            public OptionalDouble getForecastedWriteLoad(IndexMetadata indexMetadata) {
-                return indexMetadata.getForecastedWriteLoad();
-            }
-        };
+        var clusterState = stateWithStartedIndices(
+            IndexMetadata.builder("heavy-index").indexWriteLoadForecast(8.0),
+            IndexMetadata.builder("light-index-1").indexWriteLoadForecast(1.0),
+            IndexMetadata.builder("light-index-2").indexWriteLoadForecast(2.0),
+            IndexMetadata.builder("light-index-3").indexWriteLoadForecast(3.0),
+            IndexMetadata.builder("zero-write-load-index").indexWriteLoadForecast(0.0),
+            IndexMetadata.builder("no-write-load-index")
+        );
 
-        var settings = Settings.EMPTY;
+        var settings = Settings.builder()
+            // allow as many parallel rebalances as needed
+            .put(ThrottlingAllocationDecider.CLUSTER_ROUTING_ALLOCATION_NODE_CONCURRENT_RECOVERIES_SETTING.getKey(), 10)
+            .put(ThrottlingAllocationDecider.CLUSTER_ROUTING_ALLOCATION_NODE_CONCURRENT_INCOMING_RECOVERIES_SETTING.getKey(), 10)
+            .put(ThrottlingAllocationDecider.CLUSTER_ROUTING_ALLOCATION_NODE_CONCURRENT_OUTGOING_RECOVERIES_SETTING.getKey(), 10)
+            .build();
         var allocator = new BalancedShardsAllocator(
             settings,
             new ClusterSettings(settings, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS),
-            testWriteLoadForecaster
+            TEST_WRITE_LOAD_FORECASTER
         );
         var allocation = createRoutingAllocation(clusterState);
         allocator.allocate(allocation);
 
-        assertThat(allocation.metadata().getTotalNumberOfShards(), allOf(greaterThanOrEqualTo(3), lessThanOrEqualTo(5)));
         for (RoutingNode routingNode : allocation.routingNodes()) {
             var nodeIngestLoad = 0.0;
             for (ShardRouting shardRouting : routingNode) {
-                if (shardRouting.started() || shardRouting.initializing()) { // count load from the target node when relocating
+                if (shardRouting.started() || shardRouting.initializing()) { // count write load from the target node when relocating
                     var indexMetadata = clusterState.metadata().index(shardRouting.index());
-                    nodeIngestLoad += testWriteLoadForecaster.getForecastedWriteLoad(indexMetadata).orElse(0.0);
+                    nodeIngestLoad += TEST_WRITE_LOAD_FORECASTER.getForecastedWriteLoad(indexMetadata).orElse(0.0);
                 }
             }
-            assertThat(nodeIngestLoad, lessThanOrEqualTo(8.0 + 1.5));
+            assertThat(nodeIngestLoad, lessThanOrEqualTo(8.0));
```

Comment:
Implicitly this asserts that `heavy-index` is on a different node from `light-index-*` but could we be more explicit about that? Can we also assert that `zero-write-load-index` and `no-write-load-index` end up on the same node as `heavy-index` because of shard balance?